sound/soc/sof/ipc.c

```diff
@@ -798,7 +798,7 @@ int snd_sof_ipc_valid(struct snd_sof_dev *sdev)
 		return -EINVAL;
 	}
 
-	if (v->abi_version > SOF_ABI_VERSION) {
+	if (SOF_ABI_VERSION_MINOR(v->abi_version) > SOF_ABI_MINOR) {
```

Comment:
don't we need to also check the MAJOR is compatible. This feels like testing the LSB only without checking the MSB is identical. It may be premature but at some point the MAJOR is going to change

Reply:
The MAJOR revision is compared on the above line with SOF_ABI_VERSION_INCOMPATIBLE() so would be redundant here.